taxii2-feed/operations.py

```diff
@@ -0,0 +1,274 @@
+        "confidence": params.get("confidence") if params.get("confidence") is not None and params.get(
+            "confidence") != '' else 0,
+        "reputation": REPUTATION_MAP.get(params.get("reputation")) if params.get(
+            'Suspicious') is not None and params.get("Suspicious") != '' else REPUTATION_MAP.get("Suspicious"),
+        "tlp": TLP_MAP.get(params.get("tlp")) if params.get("tlp") is not None and params.get(
+            "tlp") != '' else TLP_MAP.get("White"),
+        "valid_until": max_age(params, ioc)
+    }
+
+
+def get_collections(config, params):
+    taxii = TaxiiFeed(config)
+    api_root = taxii.get_api_root_information(endpoint='taxii/')
+    response_headers = taxii.make_request(endpoint=api_root, api_info='api_root_info')
+    headers = {'Accept': response_headers['Content-Type']}
+    if 'Collection ID' in params.get('collection_type'):
+        response = taxii.make_request(endpoint=api_root + 'collections/' + str(params['collectionID']) + '/',
+                                      headers=headers)
+    else:
+        if params.get('limit') is not None or params.get('limit') != '':
+            headers.update(
+                {'Range': 'items {0}-{1}'.format(str(params.get('offset')), str(params.get('limit')))})
+        response = taxii.make_request(endpoint=api_root + 'collections/', headers=headers)
+    if response.get('collections'):
+        return response
+    else:
+        return {'collections': [response]}
+
+
+def get_objects_by_collection_id(config, params):
+    taxii = TaxiiFeed(config)
+    response = []
+    api_root = taxii.get_api_root_information(endpoint='taxii/')
+    response_headers = taxii.make_request(endpoint=api_root, api_info='api_root_info')
+    headers = {'Accept': response_headers['Content-Type']}
+    if params.get('limit') is None or params.get('limit') == '':
+        server_url = config.get('server_url')
+        if not server_url.startswith('https://'):
+            server_url = 'https://' + server_url
+        if not server_url.endswith('/'):
+            server_url += '/'
+        username = config.get('username')
+        password = config.get('password')
+        collection = Collection(
+            api_root + 'collections/' + str(params.get('collectionID')) + '/',
+            user=username, password=password)
+        for bundle in as_pages(collection.get_objects, added_after=params.get('added_after') if (
+                params.get('added_after') is not None and params.get(
+            'added_after') != '') else '1970-01-01T00:00:00.000Z', start=params.get('offset'),
+                               per_request=1000):
+            if 'objects' not in bundle.keys():
+                response.append(bundle)
+                break
+            for ioc in bundle["objects"]:
+                if ioc["type"] == "indicator" and "spec_version" in ioc.keys():
+                    response.append(stix_spec(ioc, "2.1", params))
```

Comment:
why do we need spec version in response

Reply:
format of 2.0 and 2.1 is different that's why I added that one